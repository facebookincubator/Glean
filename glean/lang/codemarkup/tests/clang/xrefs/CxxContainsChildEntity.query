query: codemarkup.cxx.CxxContainsChildEntity _
transform: [gensort, []]
