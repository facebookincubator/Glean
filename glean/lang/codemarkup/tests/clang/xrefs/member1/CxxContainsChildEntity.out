[
  "@generated",
  {
    "key": {
      "parent": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": { "name": { "key": "Foo" }, "scope": { "global_": { } } }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 17,
                "columnBegin": 1,
                "lineEnd": 25,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "foo" } } },
                  "scope": {
                    "recordWithAccess": {
                      "record": {
                        "key": {
                          "name": { "key": "Foo" },
                          "scope": { "global_": { } }
                        }
                      },
                      "access": 0
                    }
                  }
                }
              },
              "signature": {
                "key": { "returns": { "key": "void" }, "parameters": [ ] }
              },
              "method": {
                "isVirtual": false,
                "isConst": false,
                "isVolatile": false,
                "refQualifier": 0
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 20,
                "columnBegin": 3,
                "lineEnd": 24,
                "columnEnd": 3
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": { "name": { "key": "Foo" }, "scope": { "global_": { } } }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 17,
                "columnBegin": 1,
                "lineEnd": 25,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "variable": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "ptr" },
                  "scope": {
                    "recordWithAccess": {
                      "record": {
                        "key": {
                          "name": { "key": "Foo" },
                          "scope": { "global_": { } }
                        }
                      },
                      "access": 0
                    }
                  }
                }
              },
              "type": { "key": "Ptr" },
              "kind": { "field": { "mutable_": false } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 18,
                "columnBegin": 3,
                "lineEnd": 18,
                "columnEnd": 9
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": { "name": { "key": "Ptr" }, "scope": { "global_": { } } }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 9,
                "columnBegin": 1,
                "lineEnd": 13,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": {
                    "key": { "conversionOperator": { "key": "bool" } }
                  },
                  "scope": {
                    "recordWithAccess": {
                      "record": {
                        "key": {
                          "name": { "key": "Ptr" },
                          "scope": { "global_": { } }
                        }
                      },
                      "access": 0
                    }
                  }
                }
              },
              "signature": {
                "key": { "returns": { "key": "bool" }, "parameters": [ ] }
              },
              "method": {
                "isVirtual": false,
                "isConst": false,
                "isVolatile": false,
                "refQualifier": 0
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 11,
                "columnBegin": 3,
                "lineEnd": 11,
                "columnEnd": 35
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": { "name": { "key": "Ptr" }, "scope": { "global_": { } } }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 9,
                "columnBegin": 1,
                "lineEnd": 13,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "conversionOperator": { "key": "int" } } },
                  "scope": {
                    "recordWithAccess": {
                      "record": {
                        "key": {
                          "name": { "key": "Ptr" },
                          "scope": { "global_": { } }
                        }
                      },
                      "access": 0
                    }
                  }
                }
              },
              "signature": {
                "key": { "returns": { "key": "int" }, "parameters": [ ] }
              },
              "method": {
                "isVirtual": false,
                "isConst": false,
                "isVolatile": false,
                "refQualifier": 0
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 12,
                "columnBegin": 3,
                "lineEnd": 12,
                "columnEnd": 30
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": { "name": { "key": "Ptr" }, "scope": { "global_": { } } }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 9,
                "columnBegin": 1,
                "lineEnd": 13,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "variable": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "value" },
                  "scope": {
                    "recordWithAccess": {
                      "record": {
                        "key": {
                          "name": { "key": "Ptr" },
                          "scope": { "global_": { } }
                        }
                      },
                      "access": 0
                    }
                  }
                }
              },
              "type": { "key": "void *" },
              "kind": { "field": { "mutable_": false } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 10,
                "columnBegin": 3,
                "lineEnd": 10,
                "columnEnd": 13
              }
            }
          }
        }
      }
    }
  }
]